pontoon/actionlog/models.py

```diff
@@ -54,7 +56,7 @@ def validate_foreign_keys_per_action(self):
 
         if self.action_type != "translation:deleted" and (
             not self.translation or self.entity or self.locale
-        ):
+        ) and self.action_type != "comment:deleted":
```

Comment:
Aha, that's not correct. When a comment is deleted, we still want to link to the translation that is concerned. :)

Reply:
Yep, I no longer needed this check as I am passing in the associated translation (haven't pushed it yet as I'm finishing up some areas).